src/handlers/psp.ts

```diff
@@ -169,29 +168,70 @@ const initializeState = (config: Config): State => {
 const executeApiCalls = async (state: State): Promise<State> => {
   const { config, baseLogOptions, groupedSubscriptions } = state;
 
-  const apiValuePromises = groupedSubscriptions.map(({ subscriptions, template, endpoint }) => {
+  const wait = (ms: number) => new Promise((res) => setTimeout(res, ms));
+  const doRequestWithRetries = async (
+    fn: () => Promise<CallApiResult>,
+    minDelay = 1_000,
+    maxDelay = 5_000
+  ): Promise<CallApiResult> => {
+    try {
+      return await fn();
+    } catch (e) {
+      await wait(Math.floor(Math.random() * (maxDelay - minDelay + 1) + minDelay));
+      return await doRequestWithRetries(fn, minDelay, maxDelay);
+    }
```

Comment:
Can't we use the `go` function here as well? Instead of the try-catch block I mean.

Reply:
We probably can. I'll make the change and test it out.